junit/kube-api-test/client-inject/src/test/java/io/fabric8/kubeapitest/junit/sample/JUnitFabric8ClientInjectPerMethodTest.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.fabric8.kubeapitest.junit.sample;
+
+import io.fabric8.kubeapitest.junit.EnableKubeAPIServer;
+import io.fabric8.kubeapitest.junit.TestUtils;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import org.junit.jupiter.api.Test;
+
+class JUnitFabric8ClientInjectPerMethodTest {
+
+  KubernetesClient client;
+
+  @Test
+  @EnableKubeAPIServer
+  void simpleTest1() {
+    TestUtils.simpleTest(client);
+  }
+
+  @Test
+  @EnableKubeAPIServer
+  void simpleTest2() {
+    TestUtils.simpleTest(client);
```

Comment:
Instead of reusing static method for shared tests, do you think making an abstract class for simpleTest and making these classes extend that class would be better? Just like we do in httpclient tests...

Reply:
To be honest I don't really like that pattern, when we extending tests. But if everybody agrees it should be done that way in this project, I can change it.
I see it better with this "composition" over inheritance approach.

Reply:
I don't like abstract tests either. However, I don't like the usage of this static test methods :).
Anyway, I'm sure we'll find a way to refactor the tests later on, especially if they are all doing the same.